docs/tlc-config.md

```diff
@@ -0,0 +1,157 @@
+
+// Replace the constant in the left-hand side
+// with the constant expression in the right-hand side.
+assignment ::=
+    ident "=" constExpr
+
+// A constant expression that may appear in
+// the right-hand side of an assignment.
+constExpr ::=
+    modelValue
+  | integer
+  | string
+  | "{" "}"
+  | "{" constExpr ("," constExpr)* "}"
+
+// The name of a model value, see Section 14.5.3 of Specifying Systems.
+// A model value is essentially an uninterpreted constant.
+// All model values are distinct from one another. Moreover, they are
+// not equal to other values such as integers, strings, sets, etc.
+// Apalache treats model values as strings, which it declares as
+// uninterpreted constants in SMT.
+modelValue ::= ident
+
+// An integer (no bit-width assumed)
+integer ::=
+      <string matching regex [0-9]+>
+    | "-" <string matching regex [0-9]+>
+
+// A string, starts and ends with quotes,
+// a restricted set of characters is allowed (pre-UTF8 era, Paxon scripts?)
+string ::=
+    '"' <string matching regex [a-zA-Z0-9_~!@#\$%^&*-+=|(){}[\],:;`'<>.?/ ]*> '"'
+
+// Set an invariant (over unprimed variables) to be checked against
+// every reachable state.
+Invariants ::=
+    ("INVARIANT" | "INVARIANTS") ident*
+
+// Set a temporal property to be checked against the initial states.
+// Temporal properties reason about finite or infinite computations,
+// which are called behaviors in TLA+. Importantly, the computations
+// originate from the initial states.
+// APALACHE IGNORES THIS CONFIGURATION OPTION.
+Properties ::=
+    ("PROPERTY" | "PROPERTIES") ident*
+
+// Set a state predicate (over unprimed variables)
+// that restricts the state space to be explored.
+// APALACHE IGNORES THIS CONFIGURATION OPTION.
+StateConstraints ::=
+    ("CONSTRAINT" | "CONSTRAINTS") ident*
+
+// Set an action predicate (over unprimed and primed variables)
+// that restricts the transitions to be explored.
+// APALACHE IGNORES THIS CONFIGURATION OPTION.
+ActionConstraints ::=
+    ("ACTION-CONSTRAINT" | "ACTION-CONSTRAINTS") ident*
+
+// Set the name of an operator that produces a set of permutations
+// for symmetry reduction.
+// See Section 14.3.3 in Specifying Systems.
+// APALACHE IGNORES THIS CONFIGURATION OPTION.
+Symmetry ::=
+    "SYMMETRY" ident
+
+// Set the name of an operator that produces a state view
+// (some form of abstraction).
+// See Section 14.3.3 in Specifying Systems.
+// APALACHE IGNORES THIS CONFIGURATION OPTION.
+View ::=
+    "VIEW" ident
+
+// Whether the tools should check for deadlocks.
+// APALACHE IGNORES THIS CONFIGURATION OPTION.
+CheckDeadlock ::=
+    "CHECK_DEADLOCK" ("FALSE" | "TRUE")
+
+// Recent feature: https://lamport.azurewebsites.net/tla/current-tools.pdf
+// APALACHE IGNORES THIS CONFIGURATION OPTION.
+Postcondition ::=
+    "POSTCONDITION" ident
+
+// Very recent feature: https://github.com/tlaplus/tlaplus/issues/485
+// APALACHE IGNORES THIS CONFIGURATION OPTION.
+Alias ::=
+    "ALIAS" ident
```

Comment:
I doubt it's a good idea to just silently ignore so many configuration parameters. The user will expect certain behavior, and the tool will probably just fail in another place, e.g. due to missing invariant, not explaining to the user that actually his PROPERTIES definition was ignored.

What about adding to the parser smth. like two string variables, warnings, and errors? And for whatever not supported, add lines to one of those? Later, when the parsed configuration is imported, check on these, and if errors are non-empty -> abort, if warnings are non-empty -> print them out.

Reply:
Well, `ConfigurationPassImpl` warns the user about the ignored properties. However, I do not want to make the configuration model too complex, just to complain about unsupported options later. We can add warnings right in the parser. Apart from `PROPERTY`, many other ignored options are TLC optimizations, which we will most probably never support.

Reply:
Right, warnings right in the parser is better, I only was not sure whether this is possible from the architectural point of view.

Btw, there is a warning issued about PROPERTIES in the importer, but this is already outdated, the leftover from the previous ConfigParser implementation, is it right? then it should be probably deleted?

Reply:
Don't forget about [typed model values](https://tla.msr-inria.inria.fr/tlatoolbox/doc/model/model-values.html).

Reply:
I have added warnings for the options that are completely ignored by the parser such as "VIEW", "SYMMETRY", etc. We might want to extend the parser later, to propagate these options, but I do not see any value in doing that for now. The "PROPERTIES" are propagated and then ignored by the model checker with a warning, as we like to implement liveness checking later.

Reply:
> Don't forget about typed model values.

From the perspective of the config parser, there is no difference between model values and typed model values, right?